packages/permission-controller/src/Permission.ts

```diff
@@ -349,6 +352,12 @@ export type PermissionValidatorConstraint = (
   target?: string,
 ) => void;
 
+export type PermissionSideEffect = {
+  onPermitted: () => void;
```

Comment:
Curious why these first two are mandatory. Do you think we'll always need them both for any permissions that have a side-effect? I can't think of any counter-examples.

Having a JSDoc block for these would be fantastic, so that we can offer advice on what these are intended to be used for, and we can make recommendations about cleanup and so on.